From 65db9afe341b272c1aa7098502cccfecbaa10cd6 Mon Sep 17 00:00:00 2001
From: Pedro Palazon Candel <pedro@joyent.com>
Date: Fri, 18 May 2018 18:52:00 +0200
Subject: [PATCH] TRITON-397 NAT instances w/o resolvers make `sdcadm
 fix-core-vm-resolvers` fail Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
 Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 lib/steps/binder.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/steps/binder.js b/lib/steps/binder.js
index 691b880..f05904d 100644
--- a/lib/steps/binder.js
+++ b/lib/steps/binder.js
@@ -226,7 +226,7 @@ function checkCoreVmInstancesResolvers(arg, cb) {
                     func: function checkResolvers(vm, nextVm) {
                         // Binder VMs and eventually, anything we want to
                         // avoid resolvers:
-                        if (vm.resolvers.length === 0 &&
+                        if ((vm.resolvers && vm.resolvers.length === 0) &&
                             vm.internal_metadata &&
                             (vm.internal_metadata.set_resolvers === false)) {
                             nextVm();
-- 
2.21.0

